packages/fxa-auth-server/fxa-oauth-server/lib/keys.js

```diff
@@ -0,0 +1,75 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const config = require('./config');
+const JwTool = require('fxa-jwtool');
+
+const ISSUER = config.get('openid.issuer');
+
+function pub(key) {
+  // Hey, this is important. Listen up.
+  //
+  // This function pulls out only the **PUBLIC** pieces of this key.
+  // For RSA, that's the `e` and `n` values.
+  //
+  // BE CAREFUL IF YOU REFACTOR THIS. Thanks.
+  return {
+    kty: key.kty,
+    alg: 'RS256',
+    kid: key.kid,
+    'fxa-createdAt': key['fxa-createdAt'],
+    use: 'sig',
+    n: key.n,
+    e: key.e,
+  };
+}
+
+const PRIVATE_KEYS_RAW = [config.get('openid.key')];
+
+const oldPrivKey = config.get('openid.oldKey');
+if (oldPrivKey && Object.keys(oldPrivKey).length) {
+  PRIVATE_KEYS_RAW.push(oldPrivKey);
+}
+
+const SIGNING_KEY = JwTool.JWK.fromObject(config.get('openid.key'), {
+  iss: ISSUER,
+});
+
+const PUBLIC_KEYS_RAW = new Map();
+const VERIFICATION_PUBLIC_KEYS = new Map();
+
+PRIVATE_KEYS_RAW.forEach(privKeyRaw => {
+  const publicKeyRaw = pub(privKeyRaw);
+
+  PUBLIC_KEYS_RAW.set(privKeyRaw.kid, publicKeyRaw);
+
+  VERIFICATION_PUBLIC_KEYS.set(
+    privKeyRaw.kid,
+    JwTool.JWK.fromObject(publicKeyRaw, {
+      iss: ISSUER,
+    })
+  );
+});
+
+module.exports = {
+  /**
+   * Get a public JWK by `kid`. Call `.verify`
+   * w/ JWT to verify.
+   *
+   * @param {String} kid of JWK to get
+   */
+  publicJWK(kid) {
+    return VERIFICATION_PUBLIC_KEYS.get(kid);
```

Comment:
I'd be very tempted to have this throw rather than return `undefined` on error, just to make extra sure that the caller has to deal with that error path.